share/models/meta.py

```diff
@@ -1,17 +1,41 @@
+import furl
+import json
+
 from django.db import models
 from django.db import IntegrityError
 from django.contrib.postgres.fields import JSONField
 
 from share.models.base import ShareObject
 from share.models.fields import ShareForeignKey, ShareURLField, ShareManyToManyField
-from share.apps import ShareConfig as share_config
 
 
-__all__ = ('Venue', 'Award', 'Tag', 'Link', 'Subject')
+__all__ = ('Venue', 'Award', 'Tag', 'Subject', 'Identifier', 'Relation')
 
 # TODO Rename this file
 
 
+class Identifier(ShareObject):
+    # https://twitter.com/berniethoughts/
+    url = ShareURLField(unique=True)
+
+    # https://twitter.com/
+    def get_base_url(self):
+        return furl(self.url).origin
+
+
+class Relation(ShareObject):
+    with open('./share/models/relation-types.json') as fobj:
+        # TODO add label to file
+        RELATION_TYPES = [(t['key'], t['uri']) for t in json.load(fobj)]
```

Comment:
When ontologies overlap, I think it makes more sense to have a key pointing to a list of possible uris, rather than having multiple keys that are semantically equivalent.

Right now the key is stored as a value in the relation table, because it seems sufficient for the queries we'll make and avoids the overhead of joining yet another table when fetching works. The list of allowed relation types is stored in a [https://github.com/aaxelb/SHARE/blob/aee324d9b5088f039634d90eba28484b89558413/share/models/relation-types.json](json file) which is (1) used for validation and (2) exposed at `/api/relationtypes` for eventual use in the curation interface. I can change the file to have a list of uris for each key, and if/when we want to support additional ontologies we could use it in the normalizer to map a uri to a key.